### ISSUE TITLE
Can tf-ranking handle the CTR problems with high dimensional sparse data?

### ISSUE THREAD

**Issue by BloodD** [tensorflow/ranking#8]

**Comment by ramakumar1729** [comment]
Yes. One of the advantages of neural networks is that they can handle high dimensional sparse features. This is done by learning a dense representation/embedding for a given sparse feature.

TF-Ranking uses Feature Columns (see [tf.feature_column](https://www.tensorflow.org/api_docs/python/tf/feature_column)) to represent features. Look at this [unittest](https://github.com/tensorflow/ranking/blob/931e4e18d68612d0b29dc3c81994acdd4b6ab743/tensorflow_ranking/python/feature_test.py#L198) for using a combination of embedding columns and categorical columns to handle sparse data. Categorical columns take in a vocabulary as an input, or alternatively can use hash buckets to create an internal vocabulary. 

For very high dimensional sparse data, it is common to prune down the vocabulary to the top N frequently occurring feature values.

**Comment by xuanhuiwang** [comment]
tf-ranking has a loss named sigmoid_cross_entropy_loss: https://github.com/tensorflow/ranking/blob/master/tensorflow_ranking/python/losses.py#L47. If this loss is used, tf-ranking become standard regression. The sigmoid transformation of the PREDICT output corresponds to a CTR estimation. This is not the case for pairwise or listwise loss.
